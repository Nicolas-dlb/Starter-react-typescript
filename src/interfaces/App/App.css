.app {
  text-align: center;
  background-color: #282c34;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  overflow: scroll;
}

button {
  cursor: pointer;
  font-family: "Roboto";
  border: none;
}
