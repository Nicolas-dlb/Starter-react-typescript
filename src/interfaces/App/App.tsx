import React, { useState } from "react";
import "./App.css";
import Form from "../../components/form/Form";
import Header from "../../components/header/Header";
import BadRequestModal from "../../components/badRequestModal/BadRequestModal";
import useHooks from "../../utils/hooks/useHooks";
import getData from "../../utils/api/api";

function App(): JSX.Element {
  const [url, setUrl] = useState("");
  const [badRequest, setBadRequest] = useState<boolean>(false);
  const { createItem } = useHooks();

  return (
    <div className="app">
      <Header />
      <Form
        onChange={setUrl}
        onClick={() =>
          getData(url).then((data) =>
            createItem(data).catch(() => !badRequest && setBadRequest(true))
          )
        }
      />

      {badRequest && (
        <BadRequestModal
          badRequest={badRequest}
          setBadRequest={setBadRequest}
        />
      )}
    </div>
  );
}

export default App;
