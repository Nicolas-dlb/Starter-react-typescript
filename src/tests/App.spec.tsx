import { render } from "@testing-library/react";
import React from "react";
import App from "../interfaces/App/App";

describe("App", () => {
  it("Should render the app correctly", () => {
    const view = render(<App />);
    expect(view).toMatchSnapshot();
  });
});
