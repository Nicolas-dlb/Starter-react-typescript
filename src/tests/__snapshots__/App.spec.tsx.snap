// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the app correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header
          data-testid="header"
        >
          <h1>
            Klaxoon
          </h1>
        </header>
        <div
          class="form"
          data-testid="form"
        >
          <input
            aria-label="input"
            data-testid="input"
            id="input"
            tabindex="0"
            type="text"
          />
          <button
            type="button"
          >
            Add
          </button>
        </div>
        <div
          class="bookmarks"
          data-testid="bookmarksWrapper"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header
        data-testid="header"
      >
        <h1>
          Klaxoon
        </h1>
      </header>
      <div
        class="form"
        data-testid="form"
      >
        <input
          aria-label="input"
          data-testid="input"
          id="input"
          tabindex="0"
          type="text"
        />
        <button
          type="button"
        >
          Add
        </button>
      </div>
      <div
        class="bookmarks"
        data-testid="bookmarksWrapper"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
